keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -269,50 +268,39 @@ open class SqlResourceRepository(
         .insertInto(EVENT)
         .set(EVENT.UID, ULID().nextULID(event.timestamp.toEpochMilli()))
         .set(EVENT.SCOPE, event.scope)
-        .set(EVENT.REF, ref)
-        .set(EVENT.TIMESTAMP, event.timestamp)
         .set(EVENT.JSON, event)
         .execute()
     }
   }
 
   override fun delete(id: String) {
     // TODO: these should be run inside a transaction
-    val uids = sqlRetry.withRetry(READ) {
-      jooq.select(RESOURCE.UID)
-        .from(RESOURCE)
+    sqlRetry.withRetry(WRITE) {
+      jooq.deleteFrom(RESOURCE)
         .where(RESOURCE.ID.eq(id))
-        .fetch(RESOURCE.UID)
-        .map(ULID::parseULID)
+        .execute()
+        .also { count ->
+          if (count == 0) {
+            throw NoSuchResourceId(id)
+          }
+        }
     }
-
-    if (uids.isEmpty()) {
-      throw NoSuchResourceId(id)
+    sqlRetry.withRetry(WRITE) {
+      jooq.deleteFrom(EVENT)
+        .where(EVENT.SCOPE.eq(EventScope.RESOURCE))
+        .and(EVENT.REF.eq(id))
+        .execute()
     }
-
-    uids.forEach { uid ->
-      sqlRetry.withRetry(WRITE) {
-        jooq.deleteFrom(RESOURCE)
-          .where(RESOURCE.UID.eq(uid.toString()))
-          .execute()
-      }
```

Comment:
Don't need to do this in a loop any more as events will have the resource id, not one of the uids so we can delete them all at once